File: src/main/java/com/freya02/botcommands/api/annotations/CommandId.java
package com.freya02.botcommands.api.annotations;

import com.freya02.botcommands.api.BContext;
import com.freya02.botcommands.api.application.CommandPath;
import com.freya02.botcommands.api.application.GuildApplicationSettings;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Allows you to set a command ID on your application command methods.
 *
 * @see GuildApplicationSettings#getGuildsForCommandId(BContext, String, CommandPath)
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface CommandId {
	/**
	 * @return The ID of this command method, must be unique
	 */
	String value();
}

File: src/main/kotlin/io/github/freya022/botcommands/api/commands/application/annotations/CommandId.kt
package io.github.freya022.botcommands.api.commands.application.annotations

import io.github.freya022.botcommands.api.commands.application.ApplicationCommand
import io.github.freya022.botcommands.api.commands.application.slash.annotations.TopLevelSlashCommandData

/**
 * Sets a **unique** command ID on an **annotated** application command function.
 *
 * **Note:** This only applies to top-level commands, for slash commands,
 * this means the annotation needs to be used alongside [@TopLevelSlashCommandData][TopLevelSlashCommandData].
 *
 * @see ApplicationCommand.getGuildsForCommandId
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class CommandId(val value: String)
